### PR TITLE
Generate lockfile and fetch dependencies outside sandbox on retry

### PR DESCRIPTION
**Pull request by Nemo157**
for #1563

### CODE REVIEW

File: src/docbuilder/rustwide_builder.rs [hunk 1]
```diff
@@ -1019,4 +1028,19 @@ mod tests {
             Ok(())
         });
     }
+
+    #[test]
+    #[ignore]
+    fn test_locked_fails_unlocked_needs_new_deps() {
```
Comment (Nemo157) [inline src/docbuilder/rustwide_builder.rs:1034]: This is a very slow test, it takes ~2 minutes on my machine, there's probably smaller crates that could be used instead, but I don't know an easy way to find them.

Comment (jyn514) [reply src/docbuilder/rustwide_builder.rs:1034]: We could just publish one - if we prefix it with `docs_rs_test` I don't think anyone will mind.

Comment (Nemo157) [reply src/docbuilder/rustwide_builder.rs:1034]: Mmm yeah, that should be pretty easy to setup. I'll publish one and update this test tomorrow.

Comment (jyn514) [reply src/docbuilder/rustwide_builder.rs:1034]: (we could also publish to staging.crates.io, but I don't think the 1-10 crates we ever end up using are a big deal compared to the ~35k that already exist)

**Review by jyn514 — APPROVED** [review]
LGTM other than the test, feel free to merge when you've made it faster to run :)

**Comment by Nemo157** [comment]
Nice, that made the CI/build_tests about 5 minutes faster 😁

**Pull request merged**